internal/controller/controller.go

```diff
@@ -254,6 +260,36 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{RequeueAfter: r.requeueAfter}, nil
 }
 
+func (r *reconciler) redact(logger logr.Logger, obj client.Object) client.Object {
+	if len(r.pathsToRedact) == 0 {
+		return obj
+	}
+
+	logger.Info(fmt.Sprintf("redacting resource %s", r.gvk.Kind))
+	objJson, err := json.Marshal(obj)
+	if err != nil {
+		logger.Error(err, "marshalling json")
```

Comment:
why do we continue if there's an error?
I would expect to return `return obj, err` in case of an error, maybe also log it in the calling function.

Reply:
(you'll probably need to adjust the `Redact` function to also take and return a `map[string]interface{}`, but that should be fairly simple to achieve :)

Reply:
@srlk yes you are right, we've gone back and forth on this issue quite a bit and I had forgotten that we did decide to bail on error rather than send data we couldn't redact. Will change.

Will address Ramon's note too.

Reply:
@srlk I addressed your noted in a commit that I just pushed.

@tommyknows I explored yours, and unfortunately ToUnstructured only operates on struct pointers. It returns an error when passed anything other than that - even maps, which are themselves pointers. I'm not sure what the issue is with the current code, that unmarshals into an unstructured?

Reply:
Narrator: @tommyknows contributed the commit, it was good, we squashed it.